crates/red_knot_project/src/metadata/options.rs

```diff
@@ -106,8 +106,9 @@ impl Options {
             custom_typeshed: typeshed.map(|path| path.absolute(project_root, system)),
             python_path: python
                 .map(|python_path| {
-                    PythonPath::SysPrefix(python_path.absolute(project_root, system))
+                    PythonPath::from_cli_flag(python_path.absolute(project_root, system))
                 })
+                .or(PythonPath::find_virtual_env(system))
```

Comment:
Arguments to `.or` are eagerly evaluated. If we use `.or_else()`, it will be lazily evaluated:

```suggestion
                .or_else(|| PythonPath::find_virtual_env(system))
```